pkg/ensign/events.go

```diff
@@ -517,7 +517,11 @@ func (s *StreamHandler) AllowedTopics() (group *topics.NameGroup, err error) {
 			sentry.Error(s.stream.Context()).Err(err).Str("topicID", topicID.String()).Msg("could not get topic name from ID")
 			return nil, status.Errorf(codes.Internal, "could not open %s stream", s.stype)
 		}
-		group.Add(name, topicID)
+
+		if err := group.Add(name, topicID); err != nil {
+			sentry.Error(s.stream.Context()).Err(err).Str("topicID", topicID.String()).Str("topic", name).Msg("could not add topic name and ID to group")
+			return nil, status.Errorf(codes.Internal, "could not open %s stream", s.stype)
+		}
```

Comment:
I believe the error occurs when a duplicate or empty topic name or ID is added to the group. This error check should prevent the panic and instead log an error; causing the internal error to be returned to the user. 

This addresses the symptom (the panic) but might not address the root cause; what was getting added here that was causing the error in the first place? At the very least, we'll have logging information to tell us in the future.